src/Panel/Page.php

```diff
@@ -111,4 +127,73 @@ public function pickerData(array $params = []): array
             'url'         => $this->model->url()
         ]);
     }
+
+    /**
+     * Returns the data array for the
+     * view's component props
+     *
+     * @internal
+     *
+     * @return array
+     */
+    public function props(): array
+    {
+        $page = $this->model;
+
+        return array_merge(parent::props(), [
+            'model' => [
+                'content'    => $this->content(),
+                'id'         => $page->id(),
+                'parent'     => $page->parentModel()->panel()->url(true),
+                'previewUrl' => $page->previewUrl(),
+                'status'     => $page->status(),
+                'title'      => $page->title()->toString(),
+            ],
+            'next' => function () use ($page) {
+                $next = $page
+                    ->nextAll()
+                    ->filterBy('intendedTemplate', $page->intendedTemplate())
+                    ->filterBy('status', $page->status())
+                    ->filterBy('isReadable', true)
+                    ->first();
+                return $next ? $next->panel()->prevnext('title') : null;
+            },
+            'prev'   => function () use ($page) {
+                $prev = $page
+                    ->prevAll()
+                    ->filterBy('intendedTemplate', $page->intendedTemplate())
+                    ->filterBy('status', $page->status())
+                    ->filterBy('isReadable', true)
+                    ->last();
+                return $prev ? $prev->panel()->prevnext('title') : null;
+            },
+            'status' => function () use ($page) {
+                if ($status = $page->status()) {
+                    return $page->blueprint()->status()[$status] ?? null;
+                }
+            },
+        ]);
+    }
+
+    /**
+     * Returns the data array for
+     * this model's Panel routes
+     *
+     * @internal
+     *
+     * @return array
+     */
+    public function route(): array
+    {
+        $page = $this->model;
+
+        return [
+            'breadcrumb' => function () use ($page) {
+                return $page->panel()->breadcrumb();
+            },
```

Comment:
Why do we need a callback here? Can't it be `$this->breadcrumb()`?

Reply:
Yep, can just be the simple method call

Reply:
Isn't the more we put into callbacks the better? Since then this won't processed in partial reloads that don't need this key

Reply:
Ah, if that's the case, then yes.

But we shouldn't do premature optimization. If we have a use-case that actually needs page data without the breadcrumb, then it's great, but otherwise we should IMO keep it as simple as possible.

Reply:
The use case is any partial request that doesn't asks for the breadcrumb. With closures, we can first slim town the whole data array to only contain the keys required for the partial request before we actually resolve the closures to actual values. If we don't use closures, we always already resolve all the data before we slim down the array to just the partial data. Which would reduce the meaningfulness of partial requests quite a bit. That's why we should probably use as many closures as possible in the Panel data arrays.

Reply:
Yes, I understand that.

The question is: Do we *have* partial requests that specifically don't need the breadcrumb? I think we should first assume that everything is needed (which allows for the simplest syntax without the closures) and then we should optimize the props that are not always needed.

I disagree with "we should probably use as many closures as possible": Calling all these closures takes time as well, so it also slightly reduces the performance – which is bad if it turns out that a prop is always needed in our frontend implementation.

Don't get me wrong: I'm 100% for optimizing and really like that the closures are lazy. We should just not use them everywhere without considering if they are needed.

Reply:
We do, e.g. every few seconds for content locking. And we should make partial requests work best also for cases not yet used,  but e.g. used by developers. But agreed "as many closures as possible" is a bad extreme as well 😅 If it would be a straightforward return of a value, a closure is probably worse. But given that for this call we would merge flip and merge a collection and then loop across the values - and that with a page these partial content lock requests are happening every few seconds - I think here it would be worth it to keep it wrapped in a closure.

Reply:
I agree! :)